libnavigation-base/src/main/java/com/mapbox/navigation/base/logger/model/Message.kt

```diff
@@ -0,0 +1,8 @@
+package com.mapbox.navigation.base.logger.model
+
+/**
+ * Wrapper class for loggers message.
+ *
+ * @param message The message you would like logged.
+ */
+data class Message(val message: String)
```

Comment:
This wrapper seems inconvenient to me. Wouldn't below accomplish the same on the default implementation level?
```
fun v(msg: String) {
    v(null, msg, null)
}

fun v(msg: String, tr: Throwable) {
    v(null, msg, tr)
}

fun v(tag: String, msg: String) {
    v(tag, msg, null)
}

override fun v(tag: String?, msg: String, tr: Throwable?) {
    log(VERBOSE, tag, msg, tr) { Timber.v(tr, msg) }
}
```

Reply:
Discussed this internally and the idea behind it is to expose a typed message instead of a simple `String` that doesn't provide any context. This is a common practice in DDD (Domain Driven Design) and allows you to take advantage of the compiler type system power.

Linking a quote from @dpreussler recent KotlinConf awesome talk [THE POWER OF TYPES](https://speakerdeck.com/dpreussler/the-power-of-types-kotlinconf-2019)

> Don't put in name what can be in type

If you haven't already, I encourage everyone to watch it, it's 🔝 💯

Reply:
Sure, yet this is such a trivial case, that I still think it'd be more convenient to use a regular `String` on a day-to-day basis.

I'm happy to accept the PR anyway if this is yours and @Lebedinsky's preference though.